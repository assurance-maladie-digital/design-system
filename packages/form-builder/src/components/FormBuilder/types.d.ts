import { FieldList } from '../FormFieldList/types';

export interface Form {
	[sectionName: string]: FieldList;
}
