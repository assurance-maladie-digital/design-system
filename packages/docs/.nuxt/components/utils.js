// nuxt/nuxt.js#8607
export function wrapFunctional(options) {
  if (!options || !options.functional) {
    return options
  }

  const propKeys = Array.isArray(options.props) ? options.props : Object.keys(options.props || {})

  return {
    render(h) {
      const attrs = {}
      const props = {}

      for (const key in this.$attrs) {
        if (propKeys.includes(key)) {
          props[key] = this.$attrs[key]
        } else {
          attrs[key] = this.$attrs[key]
        }
      }

      return h(options, {
        on: this.$listeners,
        attrs,
        props,
        scopedSlots: this.$scopedSlots,
      }, this.$slots.default)
    }
  }
}
