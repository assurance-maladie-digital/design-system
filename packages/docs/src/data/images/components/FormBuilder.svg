<svg width="541" height="357" viewBox="0 0 541 357" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_213_106550)">
<rect width="541" height="357" fill="#E7ECF5"/>
<g filter="url(#filter0_dddddd_213_106550)">
<g clip-path="url(#clip1_213_106550)">
<rect width="780" height="56" transform="translate(62 60)" fill="#E2E7EE"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M92 94C95.3137 94 98 91.3137 98 88C98 84.6863 95.3137 82 92 82C88.6863 82 86 84.6863 86 88C86 91.3137 88.6863 94 92 94Z" fill="#ED6A5E"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M112 94C115.314 94 118 91.3137 118 88C118 84.6863 115.314 82 112 82C108.686 82 106 84.6863 106 88C106 91.3137 108.686 94 112 94Z" fill="#F4BF4F"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M132 94C135.314 94 138 91.3137 138 88C138 84.6863 135.314 82 132 82C128.686 82 126 84.6863 126 88C126 91.3137 128.686 94 132 94Z" fill="#61C554"/>
<rect x="62" y="116" width="780" height="344" fill="#F6F8FC"/>
</g>
</g>
<rect opacity="0.5" x="86" y="148" width="136" height="19" rx="9.5" fill="#CED9EB"/>
<rect x="86" y="190" width="200" height="56" rx="4" fill="white"/>
<rect x="86.5" y="190.5" width="199" height="55" rx="3.5" stroke="black" stroke-opacity="0.38"/>
<rect opacity="0.5" x="102" y="209" width="145" height="19" rx="9.5" fill="#CED9EB"/>
<rect x="86" y="269" width="200" height="56" rx="4" fill="white"/>
<rect x="86.5" y="269.5" width="199" height="55" rx="3.5" stroke="black" stroke-opacity="0.38"/>
<rect opacity="0.5" x="102" y="288" width="165" height="19" rx="9.5" fill="#CED9EB"/>
<rect x="310" y="190" width="200" height="56" rx="4" fill="white"/>
<rect x="310.5" y="190.5" width="199" height="55" rx="3.5" stroke="black" stroke-opacity="0.38"/>
<rect opacity="0.5" x="326" y="209" width="97" height="19" rx="9.5" fill="#CED9EB"/>
</g>
<defs>
<filter id="filter0_dddddd_213_106550" x="-18" y="60" width="940" height="580" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB">
<feFlood flood-opacity="0" result="BackgroundImageFix"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/>
<feOffset dy="2.76726"/>
<feGaussianBlur stdDeviation="1.1069"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.0196802 0"/>
<feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow_213_106550"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/>
<feOffset dy="6.6501"/>
<feGaussianBlur stdDeviation="2.66004"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.0282725 0"/>
<feBlend mode="normal" in2="effect1_dropShadow_213_106550" result="effect2_dropShadow_213_106550"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/>
<feOffset dy="12.5216"/>
<feGaussianBlur stdDeviation="5.00862"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.035 0"/>
<feBlend mode="normal" in2="effect2_dropShadow_213_106550" result="effect3_dropShadow_213_106550"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/>
<feOffset dy="22.3363"/>
<feGaussianBlur stdDeviation="8.93452"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.0417275 0"/>
<feBlend mode="normal" in2="effect3_dropShadow_213_106550" result="effect4_dropShadow_213_106550"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/>
<feOffset dy="41.7776"/>
<feGaussianBlur stdDeviation="16.711"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.0503198 0"/>
<feBlend mode="normal" in2="effect4_dropShadow_213_106550" result="effect5_dropShadow_213_106550"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/>
<feOffset dy="100"/>
<feGaussianBlur stdDeviation="40"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.07 0"/>
<feBlend mode="normal" in2="effect5_dropShadow_213_106550" result="effect6_dropShadow_213_106550"/>
<feBlend mode="normal" in="SourceGraphic" in2="effect6_dropShadow_213_106550" result="shape"/>
</filter>
<clipPath id="clip0_213_106550">
<rect width="541" height="357" fill="white"/>
</clipPath>
<clipPath id="clip1_213_106550">
<rect x="62" y="60" width="780" height="400" rx="16" fill="white"/>
</clipPath>
</defs>
</svg>
