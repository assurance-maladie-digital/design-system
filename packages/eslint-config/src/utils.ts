// eslint-disable-next-line ts/no-explicit-any
export function renameRules(rules: Record<string, any>, from: string, to: string) {
	return Object.fromEntries(Object.entries(rules)
		.map(([key, value]) => {
			if (key.startsWith(from)) {
				return [to + key.slice(from.length), value];
			}

			return [key, value];
		}));
}
