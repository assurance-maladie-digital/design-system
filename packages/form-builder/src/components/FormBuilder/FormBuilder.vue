<template>
	<div class="vd-form-builder w-100">
		<FormFieldList
			v-for="(section, sectionName) in form"
			:key="'section-' + sectionName"
			:list-title="section.title"
			:description="section.description"
			:fields="section.questions"
			@change="sectionUpdated($event, sectionName)"
			@refresh="$emit('refresh')"
		>
			<!--
				Pass down the scoped slots
				@see https://github.com/vuejs/vue/pull/7765 for the official discussion
				and https://stackoverflow.com/questions/50891858/vue-how-to-pass-down-slots-inside-wrapper-component/52823029#52823029
				for the source of this solution
			-->
			<template
				v-for="slot in Object.keys($scopedSlots)"
				#[slot]="scope"
			>
				<slot
					:name="slot"
					v-bind="scope"
				/>
			</template>
		</FormFieldList>
	</div>
</template>

<script lang="ts">
	import Component, { mixins } from 'vue-class-component';

	import { FormBuilderCore } from './mixins/formBuilderCore';

	import FormFieldList from '../FormFieldList';

	const MixinsDeclaration = mixins(FormBuilderCore);

	@Component({
		components: {
			FormFieldList
		}
	})
	export default class FormBuilder extends MixinsDeclaration {}
</script>

<style lang="scss" scoped>
	.vd-field-list + .vd-field-list {
		padding-top: 46px;
		border-top: solid 1px rgba(0, 0, 0, .12);

		&.theme--dark {
			border-color: rgba(255, 255, 255, .12);
		}
	}
</style>
