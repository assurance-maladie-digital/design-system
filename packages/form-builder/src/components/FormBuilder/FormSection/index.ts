import FormSection from './FormSection.vue';

export default FormSection;
