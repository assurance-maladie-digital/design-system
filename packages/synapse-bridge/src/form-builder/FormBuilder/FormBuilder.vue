<script lang="ts">
import { defineComponent } from 'vue';

import FormFieldList from '../FormFieldList';
import FormBuilderCore from './mixins/formBuilderCore';

export default defineComponent({
  components: {
    FormFieldList,
  },
  mixins: [FormBuilderCore],
  emits: ['refresh'],
});
</script>

<template>
  <div class="vd-form-builder w-100">
    <FormFieldList
      v-for="(section, sectionName) in modelValue"
      :key="`section-${sectionName}`"
      :description="section.description"
      :list-title="section.title"
      :model-value="section.questions"
      @refresh="$emit('refresh')"
      @update:model-value="sectionUpdated($event, String(sectionName))"
    >
      <!--
			Pass down the scoped slots
			@see https://github.com/vuejs/vue/pull/7765 for the official discussion
			and https://stackoverflow.com/questions/50891858/vue-how-to-pass-down-slots-inside-wrapper-component
			/52823029#52823029
			for the source of this solution
		-->
      <template v-for="slot in Object.keys($slots)" #[slot]="scope">
        <slot :name="slot" v-bind="scope" />
      </template>
    </FormFieldList>
  </div>
</template>

<style lang="scss" scoped>
.vd-field-list + .vd-field-list {
	padding-top: 46px;
	border-top: solid 1px rgba(0, 0, 0, 0.12);

	&.theme--dark {
		border-color: rgba(255, 255, 255, 0.12);
	}
}
</style>
