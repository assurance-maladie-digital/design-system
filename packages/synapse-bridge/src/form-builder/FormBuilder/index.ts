import FormBuilder from './FormBuilder.vue';

export default FormBuilder;
