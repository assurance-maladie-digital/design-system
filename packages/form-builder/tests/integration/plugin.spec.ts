import Vue from 'vue';

import FormBuilder from '../../src';

import { getComponents } from '@cnamts/vue-dot/tests/unit/integration/utils';

describe('plugin', () => {
	it('should register components', () => {
		Vue.use(FormBuilder);

		expect(getComponents(Vue)).toMatchSnapshot();
	});
});
