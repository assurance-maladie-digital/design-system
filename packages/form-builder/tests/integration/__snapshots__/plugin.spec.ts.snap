// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin should register components 1`] = `
Array [
  "FormBuilder",
  "FormField",
]
`;
